packages/mui-joy/src/Link/Link.tsx

```diff
@@ -0,0 +1,236 @@
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { OverridableComponent } from '@mui/types';
+import {
+  unstable_capitalize as capitalize,
+  unstable_useForkRef as useForkRef,
+  unstable_useIsFocusVisible as useIsFocusVisible,
+} from '@mui/utils';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import styled from '../styles/styled';
+import useThemeProps from '../styles/useThemeProps';
+import Typography from '../Typography';
+import linkClasses, { getLinkUtilityClass } from './linkClasses';
+import { LinkProps, LinkTypeMap } from './LinkProps';
+
+const useUtilityClasses = (ownerState: LinkProps) => {
+  const { component, level, color, variant, underline, focusVisible, disabled } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      color && `color${capitalize(color)}`,
+      disabled && 'disabled',
+      focusVisible && 'focusVisible',
+      level,
+      underline && `underline${capitalize(underline)}`,
+      variant && `variant${capitalize(variant)}`,
+      component === 'button' && 'button',
+    ],
+  };
+
+  return composeClasses(slots, getLinkUtilityClass, {});
+};
+
+const LinkRoot = styled(Typography, {
+  name: 'MuiLink',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})<{ ownerState: LinkProps }>(({ theme, ownerState }) => {
+  return [
+    {
+      ...(ownerState.level && ownerState.level !== 'inherit' && theme.typography[ownerState.level]),
+      ...(ownerState.underline === 'none' && {
+        textDecoration: 'none',
+      }),
+      ...(ownerState.underline === 'hover' && {
+        textDecoration: 'none',
+        '&:hover': {
+          textDecoration: 'underline',
+        },
+      }),
+      ...(ownerState.underline === 'always' && {
+        textDecoration: 'underline',
+        '&:hover': {
+          textDecorationColor: 'inherit',
+        },
+      }),
+      // Same reset as ButtonBase.root
+      ...(ownerState.component === 'button' && {
```

Comment:
the padding & margin inline does not work because of this condition. @hbjORbj

Reply:
```suggestion
```
I don't think we need this condition.

Reply:
I agree. It looks much better.